README.md

```diff
@@ -81,6 +81,15 @@ The `filter_lists/*.json` files are lists of elements, each describing a filter
 
 - `support_url` is somewhere a user can ask for help with the filter list.
 
+### Optional fields
+
+The following 4 fields are all optional and default to `false` or `0` if omitted.
+
+- `hidden` (`boolean`) should the component be displayed in brave://settings/shields/filters as an additional list option?
+- `default_enabled` (`boolean`) should the component be enabled by default?
+- `first_party_protections` (`boolean`) should first-party heuristics be applied to rules from the list?
+- `permission_mask` (`number`) what scriptlet permissions should be granted to rules from the list?
```

Comment:
I don't understand what a numeric permission value means in this context. It seems like this should be an enum

Reply:
It is a mask, as in "bitmask". There's 8 independent bits that can be set or unset. They don't necessarily have any fixed meaning at the moment, other than that we're using bit 0 for uBlock Origin's "trusted" resources and bit 1 for any Brave-specific sensitive resources.

This would indeed be good to document here.

Reply:
well, on the cpp side we generally do that with  an enum like
```
enum : uint8_t {
  kIsInactiveUser = 0,
  kIsDailyUser = (1 << 0),
  kIsWeeklyUser = (1 << 1),
  kIsMonthlyUser = (1 << 2),
};
```

Reply:
neither brave-core nor adblock-rust actually attempt to prescribe any meaning to any of the bits; they just apply bitwise operations and allow or deny permission accordingly. The permissions are both required and granted from within this repository.

Reply:
well, in any case this needs some explanation about what is going on (like what we discussed in DM) because right now it's really not clear at all

Reply:
added context to the README